# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentChild.cpp
# Commit: ab54092b8afc
# Full Hash: ab54092b8afc8e56dd3f8162344a4ce38ec45999
# Author: Niklas Goegge <ngogge@mozilla.com>
# Date: 2021-04-29 09:26:05
# Regressor Bug: 1708150
# File Overlap Count: 1
# Description:
#   Bug 1708150 - Add user activation flag to reload, goBack and goForward r=ckerschb,Gijs,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D110245
# ==============================================================================

diff -r a9a52f01de13 -r ab54092b8afc dom/ipc/ContentChild.cpp
--- a/dom/ipc/ContentChild.cpp	Wed Apr 28 11:09:38 2021 +0000
+++ b/dom/ipc/ContentChild.cpp	Wed Apr 28 11:26:49 2021 +0000
@@ -4314,7 +4314,8 @@
 
 mozilla::ipc::IPCResult ContentChild::RecvGoBack(
     const MaybeDiscarded<BrowsingContext>& aContext,
-    const Maybe<int32_t>& aCancelContentJSEpoch, bool aRequireUserInteraction) {
+    const Maybe<int32_t>& aCancelContentJSEpoch, bool aRequireUserInteraction,
+    bool aUserActivation) {
   if (aContext.IsNullOrDiscarded()) {
     return IPC_OK();
   }
@@ -4324,7 +4325,7 @@
     if (aCancelContentJSEpoch) {
       docShell->SetCancelContentJSEpoch(*aCancelContentJSEpoch);
     }
-    docShell->GoBack(aRequireUserInteraction);
+    docShell->GoBack(aRequireUserInteraction, aUserActivation);
 
     if (BrowserChild* browserChild = BrowserChild::GetFrom(docShell)) {
       browserChild->NotifyNavigationFinished();
@@ -4336,7 +4337,8 @@
 
 mozilla::ipc::IPCResult ContentChild::RecvGoForward(
     const MaybeDiscarded<BrowsingContext>& aContext,
-    const Maybe<int32_t>& aCancelContentJSEpoch, bool aRequireUserInteraction) {
+    const Maybe<int32_t>& aCancelContentJSEpoch, bool aRequireUserInteraction,
+    bool aUserActivation) {
   if (aContext.IsNullOrDiscarded()) {
     return IPC_OK();
   }
@@ -4346,7 +4348,7 @@
     if (aCancelContentJSEpoch) {
       docShell->SetCancelContentJSEpoch(*aCancelContentJSEpoch);
     }
-    docShell->GoForward(aRequireUserInteraction);
+    docShell->GoForward(aRequireUserInteraction, aUserActivation);
 
     if (BrowserChild* browserChild = BrowserChild::GetFrom(docShell)) {
       browserChild->NotifyNavigationFinished();
@@ -4358,7 +4360,7 @@
 
 mozilla::ipc::IPCResult ContentChild::RecvGoToIndex(
     const MaybeDiscarded<BrowsingContext>& aContext, const int32_t& aIndex,
-    const Maybe<int32_t>& aCancelContentJSEpoch) {
+    const Maybe<int32_t>& aCancelContentJSEpoch, bool aUserActivation) {
   if (aContext.IsNullOrDiscarded()) {
     return IPC_OK();
   }
@@ -4368,7 +4370,7 @@
     if (aCancelContentJSEpoch) {
       docShell->SetCancelContentJSEpoch(*aCancelContentJSEpoch);
     }
-    docShell->GotoIndex(aIndex);
+    docShell->GotoIndex(aIndex, aUserActivation);
 
     if (BrowserChild* browserChild = BrowserChild::GetFrom(docShell)) {
       browserChild->NotifyNavigationFinished();